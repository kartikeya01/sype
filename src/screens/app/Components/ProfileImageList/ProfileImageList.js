import React from 'react';
import { FlatList, View, StyleSheet, TouchableOpacity} from 'react-native';
import { Image } from 'react-native';
import { IconButton } from 'react-native-paper';
import Post from './../Post/Post'
 

export default class ProfileImageList extends React.Component {
 
    
    
    
    
  
    state = {
      visible: true,
      username: 'bujarsefa',
      profile_img: 'https://www.familyandmedia.eu/wp-content/uploads/2018/10/529382-4386816-selfie-psicologia-725x545.jpg',
      followers: '100',
      following: '2',
      posts: '10',
      postList: [
        {
            
            post_id: 1, 
            isLiked: false,
            isBookmarked: false,
            commentsLength: 1,
            caption: 'Quarantine selfie #corona', 
            likes: 2, 
            saves: 0, 
            username: 'bujarsefa',
            likers: 
              {
                dennyliang: true
              }, 
            bookmarkers: 
              {
                dennyliang: true
              }, 
            comments: [
              {
                userName: 'Model1234',
                userImage: 'https://images.unsplash.com/photo-1534528741775-53994a69daeb?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&w=1000&q=80',
                comment: 'Slay girllll!!!'
              }
            ], 
            
            img: 'https://www.familyandmedia.eu/wp-content/uploads/2018/10/529382-4386816-selfie-psicologia-725x545.jpg', 
            date_posted: '10-10-2019'
        },
        {
          post_id: 2,
          isLiked: false,
        isBookmarked: false,
        commentsLength: 1,
          caption: 'Quarantine selfie #corona', 
          likes: 2, 
          saves: 0, 
          username: 'bujarsefa',
          likers: 
            {
              dennyliang: true
            }, 
          bookmarkers: 
            {
              dennyliang: true
            }, 
          comments: [
            {
              userName: 'Model1234',
              userImage: 'https://images.unsplash.com/photo-1534528741775-53994a69daeb?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&w=1000&q=80',
              comment: 'Slay girllll!!!'
            }
          ], 
          
          img: 'https://www.familyandmedia.eu/wp-content/uploads/2018/10/529382-4386816-selfie-psicologia-725x545.jpg', 
          date_posted: '10-10-2019'
      },
      {
        post_id: 3,
        isLiked: false,
        isBookmarked: false, 
        commentsLength: 1,
        caption: 'Quarantine selfie #corona', 
        likes: 2, 
        username: 'bujarsefa',
        saves: 0, 
        likers: 
          {
            dennyliang: true
          }, 
        bookmarkers: 
          {
            dennyliang: true
          }, 
        comments: [
          {
            userName: 'Model1234',
            userImage: 'https://images.unsplash.com/photo-1534528741775-53994a69daeb?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&w=1000&q=80',
            comment: 'Slay girllll!!!'
          }
        ], 
        
        img: 'https://www.familyandmedia.eu/wp-content/uploads/2018/10/529382-4386816-selfie-psicologia-725x545.jpg', 
        date_posted: '10-10-2019'
    },
    {
      post_id: 4,
      isLiked: false,
        isBookmarked: false, 
        commentsLength: 1, 
      caption: 'Quarantine selfie #corona', 
      likes: 2, 
      username: 'bujarsefa',
      saves: 0, 
      likers: 
        {
          dennyliang: true
        }, 
      bookmarkers: 
        {
          dennyliang: true
        }, 
      comments: [
        {
          userName: 'Model1234',
          userImage: 'https://images.unsplash.com/photo-1534528741775-53994a69daeb?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&w=1000&q=80',
          comment: 'Slay girllll!!!'
        }
      ], 
      
      img: 'https://www.familyandmedia.eu/wp-content/uploads/2018/10/529382-4386816-selfie-psicologia-725x545.jpg', 
      date_posted: '10-10-2019'
  },
  {
    post_id: 5, 
    isLiked: false,
        isBookmarked: false, 
        commentsLength: 1,
    caption: 'Quarantine selfie #corona', 
    likes: 2,
    username: 'bujarsefa', 
    saves: 0, 
    likers: 
      {
        dennyliang: true
      }, 
    bookmarkers: 
      {
        dennyliang: true
      }, 
    comments: [
      {
        userName: 'Model1234',
        userImage: 'https://images.unsplash.com/photo-1534528741775-53994a69daeb?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&w=1000&q=80',
        comment: 'Slay girllll!!!'
      }
    ], 
    
    img: 'https://www.familyandmedia.eu/wp-content/uploads/2018/10/529382-4386816-selfie-psicologia-725x545.jpg', 
    date_posted: '10-10-2019'
},
{
  post_id: 6, 
  isLiked: false,
        isBookmarked: false, 
        commentsLength: 1,
  caption: 'Quarantine selfie #corona', 
  likes: 2, 
  username: 'bujarsefa',
  saves: 0, 
  likers: 
    {
      dennyliang: true
    }, 
  bookmarkers: 
    {
      dennyliang: true
    }, 
  comments: [
    {
      userName: 'Model1234',
      userImage: 'https://images.unsplash.com/photo-1534528741775-53994a69daeb?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&w=1000&q=80',
      comment: 'Slay girllll!!!'
    }
  ], 
  
  img: 'https://www.familyandmedia.eu/wp-content/uploads/2018/10/529382-4386816-selfie-psicologia-725x545.jpg', 
  date_posted: '10-10-2019'
},
{
    post_id: 7, 
    isLiked: false,
        isBookmarked: false, 
        commentsLength: 1,
    caption: 'Quarantine selfie #corona', 
    likes: 2, 
    username: 'bujarsefa',
    saves: 0, 
    likers: 
      {
        dennyliang: true
      }, 
    bookmarkers: 
      {
        dennyliang: true
      }, 
    comments: [
      {
        userName: 'Model1234',
        userImage: 'https://images.unsplash.com/photo-1534528741775-53994a69daeb?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&w=1000&q=80',
        comment: 'Slay girllll!!!'
      }
    ], 
    
    img: 'https://www.familyandmedia.eu/wp-content/uploads/2018/10/529382-4386816-selfie-psicologia-725x545.jpg', 
    date_posted: '10-10-2019'
  },
  {
    post_id: 8, 
    isLiked: false,
        isBookmarked: false, 
        commentsLength: 1,
    caption: 'Quarantine selfie #corona', 
    likes: 2, 
    saves: 0, 
    username: 'bujarsefa',
    likers: 
      {
        dennyliang: true
      }, 
    bookmarkers: 
      {
        dennyliang: true
      }, 
    comments: [
      {
        userName: 'Model1234',
        userImage: 'https://images.unsplash.com/photo-1534528741775-53994a69daeb?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&w=1000&q=80',
        comment: 'Slay girllll!!!'
      }
    ], 
    
    img: 'https://www.familyandmedia.eu/wp-content/uploads/2018/10/529382-4386816-selfie-psicologia-725x545.jpg', 
    date_posted: '10-10-2019'
  },
  {
    post_id: 9, 
    isLiked: false,
        isBookmarked: false, 
        commentsLength: 1,
    caption: 'Quarantine selfie #corona', 
    likes: 2, 
    saves: 0,
    username: 'bujarsefa', 
    likers: 
      {
        dennyliang: true
      }, 
    bookmarkers: 
      {
        dennyliang: true
      }, 
    comments: [
      {
        userName: 'Model1234',
        userImage: 'https://images.unsplash.com/photo-1534528741775-53994a69daeb?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&w=1000&q=80',
        comment: 'Slay girllll!!!'
      }
    ], 
    
    img: 'https://www.familyandmedia.eu/wp-content/uploads/2018/10/529382-4386816-selfie-psicologia-725x545.jpg', 
    date_posted: '10-10-2019'
  },
  {
    post_id: 10, 
    isLiked: false,
        isBookmarked: false, 
        commentsLength: 1,
    caption: 'Quarantine selfie #corona', 
    likes: 2, 
    saves: 0, 
    username: 'bujarsefa',
    likers: 
      {
        dennyliang: true
      }, 
    bookmarkers: 
      {
        dennyliang: true
      }, 
    comments: [
      {
        userName: 'Model1234',
        userImage: 'https://images.unsplash.com/photo-1534528741775-53994a69daeb?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&w=1000&q=80',
        comment: 'Slay girllll!!!'
      }
    ], 
    
    img: 'https://www.familyandmedia.eu/wp-content/uploads/2018/10/529382-4386816-selfie-psicologia-725x545.jpg', 
    date_posted: '10-10-2019'
  },
  {
    post_id: 11, 
    isLiked: false,
        isBookmarked: false, 
        commentsLength: 1,
    caption: 'Quarantine selfie #corona', 
    likes: 2, 
    saves: 0, 
    username: 'bujarsefa',
    likers: 
      {
        dennyliang: true
      }, 
    bookmarkers: 
      {
        dennyliang: true
      }, 
    comments: [
      {
        userName: 'Model1234',
        userImage: 'https://images.unsplash.com/photo-1534528741775-53994a69daeb?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&w=1000&q=80',
        comment: 'Slay girllll!!!'
      }
    ], 
    
    img: 'https://www.familyandmedia.eu/wp-content/uploads/2018/10/529382-4386816-selfie-psicologia-725x545.jpg', 
    date_posted: '10-10-2019'
  },

      ],
      
      
      profile_view: 'grid', 
      /** This is our 'global' constant which determines how the user's profile images are displayed on the screen.
       * If it is a grid layout, we can view 3 images in each row. If it is listview, we only view one.
       * Note the check occurs in function renderList and in passing to 
       */


    };



    renderGrid = (element) => {
        const styling = {
            width: '32.5%',
            margin: 1,
            overflow: 'hidden',
            borderRadius: 5
        }
        return (
            <TouchableOpacity activeOpacity= {1} style={styling} onPress = {() => {console.log(element.item.post_id)}}>
                <Image source={{ uri: element.item.img }} style={{width: '100%', aspectRatio: 1 }} />
            </TouchableOpacity>
        );
    }




    render() {
        return (
            <View style={styles.profileImageViewContainer}>
    
                <View style={styles.profileViewButtonsContainer}>
                    <IconButton style={{backgroundColor: 'white'}} icon="apps" onPress={() => this.setState({profile_view : 'grid'})} />
                    <IconButton style={{backgroundColor: 'white'}} icon="view-headline" onPress={() => this.setState({profile_view : 'list'})  }/>
                </View>
            
                
                { this.state.profile_view == 'grid' ? 
                
                    <FlatList
                        key= {this.state.profile_view} 
                        data = {this.state.postList}
                        renderItem = {element => this.renderGrid(element) }
                        keyExtractor = {element => element}
                        numColumns={3}
                        extraData={this.state.profile_view}
                    />
                
                    :
                    /** NOTE, we must directly pass in each prop. Otherwise, it will issue an error. */
                    <FlatList
                    key= {this.state.profile_view} 
                    data = {this.state.postList}
                    renderItem = {element => <Post img = {element.item.img} commentsLength = {element.item.commentsLength} caption = {element.item.caption} likes = {element.item.likes} isLiked={element.item.isLiked} isBookmarked={element.item.isBookmarked} username={element.item.username}/> }
                    keyExtractor = {element => element}
                    numColumns={1}
                    extraData={this.state.profile_view}
                    />

                    /** Also note, i originaly had conditions on both numColum + renderItem function
                      * which would render a different list based on the variable profile_view.
                      * But, I'm not sure which one is the 'better' solution + why. */

                
                
                }


                

            </View>




      );
    }
  }


const styles = StyleSheet.create({
  profileViewButtonsContainer: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'center',
  },
  profileImageViewContainer: {
      flex: 1, 
      backgroundColor: 'white'
  }
});