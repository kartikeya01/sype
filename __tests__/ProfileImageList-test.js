import 'react-native';
import React from 'react';
import ProfileImageList from '@screens/app/Components/ProfileImageList/ProfileImageList';
import renderer from 'react-test-renderer';
import {shallow} from 'enzyme';
import {IconButton} from 'react-native-paper';
import {FlatList} from 'react-native';

/** Global array used to not throw errors for tests that must have predefined values. */
const postListPreDefinedToWork = [
  {
    username: '',
    commentsLength: 0,
    caption: '',
    likes: 0,
    isLiked: true,
    isBookmarked: true,
    img: '',
  },
];

/** ======== SNAPSHOT TEST =========== */
test('Profile Image List snapshot test', () => {
  /**  ProfileImageList has no passed in props. By default profileview state is 'grid' */
  const treeGrid = renderer.create(<ProfileImageList />).toJSON();
  expect(treeGrid).toMatchSnapshot();
  const treeEmptyArray = renderer.create(<ProfileImageList postList={[]} />)
    .toJSON;
  expect(treeEmptyArray).toMatchSnapshot();
  const treePreDefinedVals = renderer.create(
    <ProfileImageList postList={postListPreDefinedToWork} />,
  ).toJSON;
  expect(treePreDefinedVals).toMatchSnapshot();
});

/** ======= HELPER FUNCTIONS ======== */

/** Function: testIconButtonClicked()
 * Test for ensuring that the profile_view state changes when the appropriate icon view button is selected.
 * Note the onPress is a sub-child component of the ProfileImageList component.
 * Clicking app icon button changes profile_view state to 'grid'
 * Clicking view-headline icon button changes profile_view state to 'list'
 */
function testIconButtonClicked() {
  describe('Test for profile_view state change based on associated icon', () => {
    it('onAppsIconButtonClicked profile_view state is grid', () => {
      const wrapper = shallow(
        <ProfileImageList postList={postListPreDefinedToWork} />,
      );
      const profileImageListInstance = wrapper.instance(); /** To ensure they all use the same instance declare above */
      const iconButtonWrapper = wrapper.find(IconButton);
      const appsIconButtonWrapper = iconButtonWrapper.find({
        icon: 'apps',
      }); /** Next find the one that has the property icon grid. */
      const appsIconButtonProps = appsIconButtonWrapper.first().props();
      appsIconButtonProps.onPress(); /** Click the onPress function which changes the state. */
      /** We only care that the state is changing as needed. and not what it was before */
      const profileViewStateAfterClick =
        profileImageListInstance.state.profile_view;
      expect(profileViewStateAfterClick).toEqual('grid');
    });

    it('onViewHeadlineIconButtonClicked profile_view state is list', () => {
      const wrapper = shallow(
        <ProfileImageList postList={postListPreDefinedToWork} />,
      );
      const profileImageListInstance = wrapper.instance(); /** To ensure they all use the same instance declare above. */
      const iconButtonWrapper = wrapper.find(IconButton);
      const viewHeadlineIconButtonWrapper = iconButtonWrapper.find({
        icon: 'view-headline',
      }); /** Next find the one that has the property icon grid. */
      const viewHeadlineIconButtonProps = viewHeadlineIconButtonWrapper
        .first()
        .props();
      viewHeadlineIconButtonProps.onPress(); /** Click the onPress function which changes the state. */
      /** We only care that the state is changing as needed. and not what it was before */
      const profileViewStateAfterClick =
        profileImageListInstance.state.profile_view;
      expect(profileViewStateAfterClick).toEqual('list');
    });
  });
}

/** Funciton: testWhichFlatlistRenders
 * The profile_view dictates how many columns the flatlist that render has.
 * A grid profile view will have 3 column flat list rendered (and not 1)
 * A list profile view will have 1 column flatlist rendered (and not 3)
 */
function testWhichFlatlistRenders() {
  describe('Test the rendered flatlist based on profile_view state', () => {
    it('Profile View state = grid and 3 column flatlist is present', () => {
      const wrapper = shallow(
        <ProfileImageList postList={postListPreDefinedToWork} />,
      );
      const instance = wrapper.instance();
      instance.setState({
        profile_view: 'grid',
      }); /** WE just want to check the grid view first. (Force value) */
      const flatlistWrapper = wrapper.find(FlatList);
      expect(flatlistWrapper.find({numColumns: 3})).toHaveLength(1);
    });

    it('Profile View state = grid and 2 column flatlist is NOT present', () => {
      const wrapper = shallow(
        <ProfileImageList postList={postListPreDefinedToWork} />,
      );
      const instance = wrapper.instance();
      instance.setState({
        profile_view: 'grid',
      }); /** WE just want to check the grid view first. (Force value) */

      const flatlistWrapper = wrapper.find(FlatList);
      expect(flatlistWrapper.find({numColumns: 1})).toHaveLength(0);
    });

    it('Profile View state = list and 1 column flatlist is present', () => {
      const wrapper = shallow(
        <ProfileImageList postList={postListPreDefinedToWork} />,
      );
      const instance = wrapper.instance();
      instance.setState({
        profile_view: 'list',
      }); /** WE just want to check the list view first. (Force value) */
      /** get the current state -- For what reason? */
      const flatlistWrapper = wrapper.find(FlatList);
      expect(flatlistWrapper.find({numColumns: 1})).toHaveLength(1);
    });

    it('Profile View state = list and 3 column flatlist is NOT present', () => {
      const wrapper = shallow(
        <ProfileImageList postList={postListPreDefinedToWork} />,
      );
      const instance = wrapper.instance();
      instance.setState({
        profile_view: 'list',
      }); /** WE just want to check the list view first. (Force value) */

      const flatlistWrapper = wrapper.find(FlatList);
      expect(flatlistWrapper.find({numColumns: 3})).toHaveLength(0);
    });
  });
}

/** PropTypes Tests! Each ProfileImageList has the posts and postchildren prop tests */

function testPropTypesWithoutErrors(postList) {
  it('testing FlatList data (No errors) where postList is ' + postList, () => {
    const wrapper = shallow(<ProfileImageList postList={postList} />);
    const flatList = wrapper.find(FlatList).at(0);
    const flatListData = flatList.props().data;
    expect(flatListData).toEqual(postList);
  });
}

/** Precondition, all variables must be defined.
 * ComponentName, is the actual imported component, NOT the string name (not sure if it will work that way!).
 * props is a {key: value}. Ensure all props meet the requirements, EXCEPT the one you want to test for errors,
 * otherwise it will give errors for other props.
 *
 *
 * Need to get back to this in the next sprint or iteration. This is becasue I want to check the proptypes of postList which itself is a proptype.
 * So, I am not quite sure how to define postList first, and then check its subchildren.
 * Therefore, will get back to this.
 *
 */
// function testPropTypesWithErrors(componentName, props, propChecking, errorMsg, testDescription) {
//   it('Throws failed propType Error for ' + testDescription, () => {
//     let result = checkPropTypes(
//       componentName.propTypes,
//       props,
//       'prop',
//       componentName.propChecking,
//     );
//     assert(result === errorMsg);
//   });
// }

/** Precondition: Input must be an array either [] or that achieves the PropTypes */
function testingPostListIsArrayInputToBeTrue(postList) {
  it('testing postList is array', () => {
    const wrapper = shallow(<ProfileImageList postList={postList} />);
    const instance = wrapper.instance();
    const postListProp = instance.props.postList;
    expect(Array.isArray(postListProp)).toBe(true);
  });
}

/** ======= TEST CASES ======= */
testIconButtonClicked();
testWhichFlatlistRenders();
testPropTypesWithoutErrors(postListPreDefinedToWork);
testPropTypesWithoutErrors([]);

testingPostListIsArrayInputToBeTrue(postListPreDefinedToWork);
testingPostListIsArrayInputToBeTrue([]);

// Just to verify that this will return false
//testingPostListIsArrayInputToBeTrue('');

/** To get back to... */
// testPropTypesWithErrors(ProfileImageList, {username: undefined}, 'username', 'error', 'testing username not string');
