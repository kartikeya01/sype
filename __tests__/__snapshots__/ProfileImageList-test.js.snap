// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile Image List snapshot test 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <View
      accessibilityRole="button"
      accessibilityStates={Array []}
      accessible={true}
      focusable={true}
      hitSlop={
        Object {
          "bottom": 6,
          "left": 6,
          "right": 6,
          "top": 6,
        }
      }
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Array [
            Object {
              "alignItems": "center",
              "justifyContent": "center",
              "margin": 6,
              "overflow": "hidden",
            },
            Object {
              "borderRadius": 18,
              "height": 36,
              "width": 36,
            },
            undefined,
            Object {
              "backgroundColor": "white",
            },
          ],
        ]
      }
    >
      <View
        style={null}
      >
        <Text
          accessibilityElementsHidden={true}
          allowFontScaling={false}
          importantForAccessibility="no-hide-descendants"
          pointerEvents="none"
          style={
            Array [
              Object {
                "color": "#000000",
                "fontSize": 24,
              },
              Array [
                Object {
                  "transform": Array [
                    Object {
                      "scaleX": 1,
                    },
                  ],
                },
                Object {
                  "backgroundColor": "transparent",
                },
              ],
              Object {
                "fontFamily": "Material Design Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
      </View>
    </View>
    <View
      accessibilityRole="button"
      accessibilityStates={Array []}
      accessible={true}
      focusable={true}
      hitSlop={
        Object {
          "bottom": 6,
          "left": 6,
          "right": 6,
          "top": 6,
        }
      }
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Array [
            Object {
              "alignItems": "center",
              "justifyContent": "center",
              "margin": 6,
              "overflow": "hidden",
            },
            Object {
              "borderRadius": 18,
              "height": 36,
              "width": 36,
            },
            undefined,
            Object {
              "backgroundColor": "white",
            },
          ],
        ]
      }
    >
      <View
        style={null}
      >
        <Text
          accessibilityElementsHidden={true}
          allowFontScaling={false}
          importantForAccessibility="no-hide-descendants"
          pointerEvents="none"
          style={
            Array [
              Object {
                "color": "#000000",
                "fontSize": 24,
              },
              Array [
                Object {
                  "transform": Array [
                    Object {
                      "scaleX": 1,
                    },
                  ],
                },
                Object {
                  "backgroundColor": "transparent",
                },
              ],
              Object {
                "fontFamily": "Material Design Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
      </View>
    </View>
  </View>
  <RCTScrollView
    disableVirtualization={false}
    extraData="grid"
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={3}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View />
  </RCTScrollView>
</View>
`;

exports[`Profile Image List snapshot test 2`] = `[Function]`;

exports[`Profile Image List snapshot test 3`] = `[Function]`;
